App/Repositories/Eloquent/RoleRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Repositories\Eloquent;
+
+use App\Repositories\EloquentRepositoryInterface;
+use App\Models\Role;
+use App\Models\Permission;
+use Core\QueryBuilder;
+
+class RoleRepository implements EloquentRepositoryInterface
+{
+
```

Comment:
In Repositories to minimize queries executed in case you access some relational data eg. the permissions of a role $role->permissions, you can add this

```
private $with = ['permissions'];
```

and go on each Role:: and do this

```
Role::with($this->with)->...
```

This way laravel will build one query and do something like select * from permissions p where role_id in (1,2,3etc) (assuming multiple roles are accessed).

Do this on all repositories.

Reply:
done